src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -92,114 +95,273 @@ private async Task GetProjectMetadataAsync(
-                        {
-                            WarnForHttpSources(listPackageArgs, projectModel);
-                            var metadata = await GetPackageMetadataAsync(frameworks, listPackageArgs);
-                            await UpdatePackagesWithSourceMetadata(frameworks, metadata, listPackageArgs);
-                        }
+                    var metadata = await GetPackageMetadataAsync(frameworks, listPackageArgs);
+                    await UpdatePackagesWithSourceMetadata(frameworks, metadata, listPackageArgs);
+                }
 
-                        bool printPackages = FilterPackages(frameworks, listPackageArgs);
-                        printPackages = printPackages || ReportType.Default == listPackageArgs.ReportType;
-                        if (printPackages)
-                        {
-                            var hasAutoReference = false;
-                            List<ListPackageReportFrameworkPackage> projectFrameworkPackages = ProjectPackagesPrintUtility.GetPackagesMetadata(frameworks, listPackageArgs, ref hasAutoReference);
-                            projectModel.TargetFrameworkPackages = projectFrameworkPackages;
-                            projectModel.AutoReferenceFound = hasAutoReference;
-                        }
-                        else
-                        {
-                            projectModel.TargetFrameworkPackages = new List<ListPackageReportFrameworkPackage>();
-                        }
-                    }
+                bool printPackages = FilterPackages(frameworks, listPackageArgs) || ReportType.Default == listPackageArgs.ReportType;
+
+                if (printPackages)
+                {
+                    var hasAutoReference = false;
+                    List<ListPackageReportFrameworkPackage> projectFrameworkPackages = ProjectPackagesPrintUtility.GetPackagesMetadata(frameworks, listPackageArgs, ref hasAutoReference);
+                    projectModel.TargetFrameworkPackages = projectFrameworkPackages;
+                    projectModel.AutoReferenceFound = hasAutoReference;
                 }
                 else
                 {
-                    projectModel.AddProjectInformation(ProblemType.Error,
-                        string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_ErrorReadingAssetsFile, assetsPath));
+                    projectModel.TargetFrameworkPackages = new List<ListPackageReportFrameworkPackage>();
                 }
+            }
+        }
 
-                // Unload project
-                ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+        private static async Task GetVulnerabilitiesFromAuditSourcesAsync(
+            ListPackageArgs listPackageArgs,
+            ListPackageReportModel listPackageReportModel,
+            ListPackageProjectModel projectModel,
+            List<FrameworkPackages> frameworks)
+        {
+            var vulnerabilities = await GetVulnerabilityData(
```

Comment:
rule 10 in https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md. 

It's not obvious what the type is here.